gitpod-io/gitpod
# feat(gitpod-cli): add top cmd

## Description
Introducing a new command for gitpod-cli: `$ gp top`

```shell
Command line interface for Gitpod

Usage:
  gp [command]

Available Commands:
...
  top                 Display workspace resource (CPU/Memory usage)
```

## Related Issue(s)

Fixes #9521

## How to test
1. Start this workspace in prev env"
2. Run `gp top`, observe the cpu/memory usage
3. To raise the CPU usage, run: `yes > /dev/null &` once,
4. Run `gp top` again and obsere the CPU usage. Run the command above a few times, until the color of CPU changes.
5. To get the CPU back to a lower value, run: `killall yes`

## Release Notes

```release-note
Added a new command to gitpod-cli: "gp top" which displays workspace resources (CPU/Memory usage)
```
<img width="320" alt="Screenshot 2022-06-09 at 21 59 49" src="https://user-images.githubusercontent.com/2318450/172982711-258506e5-119d-4a31-8edd-2afbffa3f8cd.png">

<img width="336" alt="Screenshot 2022-06-09 at 21 59 20" src="https://user-images.githubusercontent.com/2318450/172982730-12c70634-5136-4f28-97e3-481ece124a61.png">

<img width="333" alt="Screenshot 2022-06-09 at 21 59 33" src="https://user-images.githubusercontent.com/2318450/172982741-8c2abc11-29d9-4522-af14-7f6823d75a47.png">


## Documentation
